proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -0,0 +1,1232 @@
+
+  /// Constructs a new span over the bytes within the supplied range of
+  /// positions within this span.
+  ///
+  /// The returned span's first byte is always at offset 0; unlike buffer
+  /// slices, extracted spans do not generally share their indices with the
+  /// span from which they are extracted.
+  ///
+  /// This function does not validate `bounds`; this is an unsafe operation.
+  ///
+  /// - Parameter bounds: A valid range of positions. Every position in
+  ///     this range must be within the bounds of this `RawSpan`.
+  ///
+  /// - Returns: A `Span` over the bytes within `bounds`
+  public func extracting(
+    uncheckedBounds bounds: some RangeExpression<Int>
+  ) -> Self
+
+  /// Constructs a new span over all the bytes of this span.
+  ///
+  /// The returned span's first byte is always at offset 0; unlike buffer
+  /// slices, extracted spans do not generally share their indices with the
+  /// span from which they are extracted.
+  ///
+  /// - Returns: A `RawSpan` over all the items of this span.
+  public func extracting(_: UnboundedRange) -> Self
+
+  /// Returns a span containing the initial bytes of this span,
+  /// up to the specified maximum byte count.
+  ///
+  /// If the maximum length exceeds the length of this span,
+  /// the result contains all the bytes.
+  ///
+  /// - Parameter maxLength: The maximum number of bytes to return.
+  ///   `maxLength` must be greater than or equal to zero.
+  /// - Returns: A span with at most `maxLength` bytes.
+  public func extracting(first maxLength: Int) -> Self
+
+  /// Returns a span over all but the given number of trailing bytes.
+  ///
+  /// If the number of elements to drop exceeds the number of elements in
+  /// the span, the result is an empty span.
+  ///
+  /// - Parameter k: The number of bytes to drop off the end of
+  ///   the span. `k` must be greater than or equal to zero.
+  /// - Returns: A span leaving off the specified number of bytes at the end.
+  public func extracting(droppingLast k: Int) -> Self
+
+  /// Returns a span containing the trailing bytes of the span,
+  /// up to the given maximum length.
+  ///
+  /// If the maximum length exceeds the length of this span,
+  /// the result contains all the bytes.
+  ///
+  /// - Parameter maxLength: The maximum number of bytes to return.
+  ///   `maxLength` must be greater than or equal to zero.
+  /// - Returns: A span with at most `maxLength` bytes.
+  public func extracting(last maxLength: Int) -> Self
+
+  /// Returns a span over all but the given number of initial bytes.
+  ///
+  /// If the number of elements to drop exceeds the number of bytes in
+  /// the span, the result is an empty span.
+  ///
+  /// - Parameter k: The number of bytes to drop from the beginning of
+  ///   the span. `k` must be greater than or equal to zero.
+  /// - Returns: A span starting after the specified number of bytes.
+  public func extracting(droppingFirst k: Int = 1) -> Self
```

Comment:
Should we have these convenience overloads on Span too?

Reply:
They are intended to be on `Span` too; if I forgot to add them to the proposal I will fix that.

Reply:
I'd argue to drop them entirely, and to try to live with range expressions only.

(If it turns out we do need some of them, we should add them on all related types, including the buffer pointer family.)

Reply:
Let's see if they get feedback in the updated pitch (I'm curious), and we can skip them when it comes up for review.